core/mondoo-okta-security.mql.yaml

```diff
@@ -1,379 +1,1985 @@
+              send_email_for_password_changed_enabled  = true
+            }
+            ```
+
+  - uid: mondoo-okta-security-password-changed-notifications-for-end-users
+    title: Enable password changed notification for end users
+    impact: 80
+    docs: 
       desc: |
-        While MFA is the ultimate way to add additional security and defend against credential-based attacks, not all MFA factors are secure.
+        When enabled, end users are sent an email notification to inform them that the password for their account has changed. This email contains details such as the time and location of the password change.
 
-        Traditionally popular factors like "security questions" and SMS have proven problematic. Bad actors can easily find the answers to security questions online, and SMS can leave users vulnerable to phishing and SIM jacking attacks.
+        ### Okta Recommends 
 
-        To increase security, require users to authenticate using strong factors like software-based authenticators. Ensure no one uses phone or email authenticators.
-      audit: |
-        __Use `cnspec run`__
+        Enable this email notification to inform end users when their password on their account has been changed or reset.
 
-        Run this query:
+        ### Security impact 
 
-          ```bash
-          cnspec run okta -c "okta.policies.accessPolicy.where(name == "MFA").all(rules.all(actions["appSignOn"]["verificationMethod"]["constraints"] { _["possession"]["deviceBound"] == "REQUIRED"} ))" --organization <YOUR-ORG> --token <OKTA_TOKEN>
-          ```
+        High
 
-        __Use the cnspec shell__
+        ### User impact 
 
-        1. Launch the cnspec shell:
+        Low
 
-          ```bash
-          cnspec shell okta ---organization <YOUR-ORG> --token <OKTA_TOKEN>
-          ```
+        End users are sent an email notification if they change or reset the password on their account. Password changed notifications aren't sent if the admin sets a temporary password for the account, changes the password by API or if the end user is in an inactive state.
+    variants: 
+      - uid: mondoo-okta-security-password-changed-notifications-for-end-users-runtime
+      - uid: mondoo-okta-security-password-changed-notifications-for-end-users-terraform-hcl
+      - uid: mondoo-okta-security-password-changed-notifications-for-end-users-terraform-plan
+  - uid: mondoo-okta-security-password-changed-notifications-for-end-users-runtime
+    title: Enable password changed notification for end users
+    impact: 80
+    filters: asset.platform == "okta-org"
+    mql: okta.organization.securityNotificationEmails['sendEmailForPasswordChangedEnabled'] == true
+    docs:
+      remediation: 
+        - id: console 
+          desc: |
+            ### Configure factor reset notifications
+            1. In the Admin Console, go to **Security > General**.
+            2. Under **Security Notification Emails**, click **Edit**.
```

Comment:
```suggestion
            2. Under **Security Notification Emails**, select **Edit**.
```

Reply:
@tas50 these are not our steps. they are from Okta's docs. They are copy/pasted.